projects/js-packages/publicize-components/src/social-store/reducer/connection-data.js

```diff
@@ -53,6 +55,38 @@ const connectionData = ( state = {}, action ) => {
 				creatingConnection: action.creating,
 			};
 
+		case UPDATE_CONNECTION:
+			return {
+				...state,
+				connections: state.connections.map( connection => {
+					// If the connection has a connection_id, then give it priority.
+					// Otherwise, use the id.
+					const isTargetConnection = connection.connection_id
+						? connection.connection_id === action.connectionId
+						: connection.id === action.connectionId;
```

Comment:
What does it depend on if `connection_id` is present?

Reply:
In some old code, there was `id` instead of `connection_id`